packages/trpc/server/routers/viewer/organizations/adminUpdate.handler.ts

```diff
@@ -76,10 +64,15 @@ export const adminUpdateHandler = async ({ input }: AdminUpdateOptions) => {
           isAdminAPIEnabled: !!(
             organizationSettings?.isAdminAPIEnabled ?? existingOrg.organizationSettings?.isAdminAPIEnabled
           ),
-        },
-      });
-    }
-    return updatedOrganization;
+        }
+      : undefined;
+
+  const updatedOrganization = await organizationRepository.adminUpdateForAdminPanel({
```

Comment:
A repository shouldn't contain business logic. This could be a candidate for AdminOrgUpdateService or something similar as Services are the one that know about business logic and delegate to Repository when a query needs to be run